builtin/providers/nsone/resource_monitoringjob_test.go

```diff
@@ -0,0 +1,203 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMonitoringJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccMonitoringJobBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMonitoringJobState("name", "terraform test"),
+					testAccCheckMonitoringJobExists("nsone_monitoringjob.foobar", &mj),
+					testAccCheckMonitoringJobAttributes(&mj),
+				),
+			},
+		},
+	})
+}
+
+func TestAccMonitoringJob_updated(t *testing.T) {
+	var mj nsone.MonitoringJob
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckMonitoringJobDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccMonitoringJobBasic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMonitoringJobState("name", "terraform test"),
+					testAccCheckMonitoringJobExists("nsone_monitoringjob.foobar", &mj),
+					testAccCheckMonitoringJobAttributes(&mj),
+				),
+			},
+			resource.TestStep{
+				Config: testAccMonitoringJobUpdated,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckMonitoringJobState("name", "terraform test"),
+					testAccCheckMonitoringJobExists("nsone_monitoringjob.foobar", &mj),
+					testAccCheckMonitoringJobAttributesUpdated(&mj),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckMonitoringJobState(key, value string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources["nsone_monitoringjob.foobar"]
+		if !ok {
+			return fmt.Errorf("Not found: %s", "nsone_monitoringjob.foobar")
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No ID is set")
+		}
+
+		p := rs.Primary
+		if p.Attributes[key] != value {
+			return fmt.Errorf(
+				"%s != %s (actual: %s)", key, value, p.Attributes[key])
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckMonitoringJobExists(n string, monitoringJob *nsone.MonitoringJob) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("NoID is set")
+		}
+
+		client := testAccProvider.Meta().(*nsone.APIClient)
+
+		foundMj, err := client.GetMonitoringJob(rs.Primary.Attributes["id"])
+
+		p := rs.Primary
+
+		if err != nil {
+			return err
+		}
+
+		if foundMj.Id != p.Attributes["id"] {
+			return fmt.Errorf("Monitoring Job not found")
+		}
+
+		*monitoringJob = foundMj
+
+		return nil
+	}
+}
+
+func testAccCheckMonitoringJobDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*nsone.APIClient)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "nsone_monitoringjob" {
+			continue
+		}
+
+		_, err := client.GetMonitoringJob(rs.Primary.Attributes["id"])
+
+		if err == nil {
+			return fmt.Errorf("Record still exists")
```

Comment:
I changes this to 

```
		mj, err := client.GetMonitoringJob(rs.Primary.Attributes["id"])

		if err == nil {
			return fmt.Errorf("Monitoring Job still exists %v: %#v", err, mj)
		}
```

this let me see the error message:

```
                Error: Check failed: Monitoring Job still exists <nil>: nsone.MonitoringJob{Id:"", Config:map[string]interface {}(nil), Status:map[string]nsone.MonitoringJobStatus(nil), Rules:[]nsone.MonitoringJobRule(nil), JobType:"", Regions:[]string(nil), Active:false, Frequency:0, Policy:"", RegionScope:"", Notes:"", Name:"", NotifyRepeat:0, RapidRecheck:false, NotifyDelay:0, NotifyList:"", NotifyRegional:false, NotifyFailback:false}
```

looks like there's a bug with `GetMonitoringJob()` if it's supposed to return an error on 404